### PR TITLE
CAML-MIMIC updated for Google Colab

### PR DESCRIPTION
**Pull request by chrispebble**
- created link-to-mimic3-files.sh to create data symlinks
- added saved_models dir
- changed script from linking to mimic data files to copying.  Deleted many MIMIC2 directories and references.
- fixed errors caused by incompatibilities with pytorch 0.3.0, created run_predictions notebook to test models
- fixed mimic3 directory in constants.py for colab settings
- run_predictions.ipynb updated after successful runs to show results
- run_predictions.ipynb now running CAML, CNN, GRU (50 and full for each) on colab, results saved in notebook

### CODE REVIEW

File: learn/models.py [hunk 1]
```diff
@@ -5,7 +5,8 @@
 import torch
 import torch.nn as nn
 import torch.nn.functional as F
-from torch.nn.init import xavier_uniform
+# from torch.nn.init import xavier_uniform
```
Comment (dmcguire81) [inline learn/models.py:8]: You can rely on `git` to track the diffs for you, here, if you want to take the commented lines out.

File: learn/training.py [hunk 2]
```diff
@@ -332,7 +339,7 @@ def test(model, Y, epoch, data_path, fold, gpu, version, code_inds, dicts, sampl
     parser.add_argument("--test-model", type=str, dest="test_model", required=False, help="path to a saved model to load and evaluate")
     parser.add_argument("--criterion", type=str, default='f1_micro', required=False, dest="criterion",
                         help="which metric to use for early stopping (default: f1_micro)")
-    parser.add_argument("--patience", type=int, default=3, required=False,
+    parser.add_argument("--patience", type=int, default=3, required=False, #dest="patience",
```
Comment (dmcguire81) [inline learn/training.py]: Looks like cruft, maybe.

File: notebooks/dataproc_mimic_III.ipynb [hunk 3]
```diff
@@ -343,15 +397,24 @@
...
         "id": "w4tDxreQ6Ysd",
         "outputId": "b2bfe18b-da7e-4708-b068-549c8f0b2c78"
       },
-      "outputs": [],
+      "outputs": [
+        {
+          "name": "stderr",
+          "output_type": "stream",
+          "text": [
+            "/Users/chrispebble/opt/anaconda3/envs/py37/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2878: DtypeWarning: Columns (4) have mixed types.Specify dtype option on import or set low_memory=False.\n",
```
Comment (dmcguire81) [inline notebooks/dataproc_mimic_III.ipynb]: Consider clearing all output on the notebook before saving, for two reasons:
* References to your local machine's file system, as above, are not portable
* On the off chance something in here is considered sensitive

File: notebooks/run_predictions.ipynb [hunk 4]
```diff
@@ -0,0 +1 @@
...eq length 212]\tLoss: 0.069467\n","23it [00:00, 30.38it/s]Train epoch: 145 [batch #25, batch_size 16, seq length 571]\tLoss: 0.023478\n","48it [00:01, 28.52it/s]Train epoch: 145 [batch #50, batch_size 16, seq length 709]\tLoss: 0.021578\n","74it [00:02, 28.65it/s]Train epoch: 145 [batch #75, batch_size 16, seq length 806]\tLoss: 0.032496\n","100it [00:03, 28.04it/s]Train epoch: 145 [batch #100, batch_size 16, seq length 892]\tLoss: 0.026319\n","124it [00:04, 26.56it/s]Train epoch: 145 [batch #125, batch_size 16, seq length 978]\tLoss: 0.025248\n","148it [00:05, 26.38it/s]Train epoch: 145 [batch #150, batch_size 16, seq length 1063]\tLoss: 0.029083\n","175it [00:06, 25.30it/s]Train epoch: 145 [batch #175, batch_size 16, seq length 1137]\tLoss: 0.029453\n","199it [00:07, 25.16it/s]Train epoch: 145 [batch #200, batch_size 16, seq length 1217]\tLoss: 0.028557\n","223it [00:08, 24.10it/s]Train epoch: 145 [batch #225, batch_size 16, seq length 1299]\tLoss: 0.026255\n","250it [00:09, 24.23it/s]Train epoch: 145 [batch #250, batch_size 16, seq length 1383]\tLoss: 0.027134\n","274it [00:10, 23.04it/s]Train epoch: 145 [batch #275, batch_size 16, seq length 1470]\tLoss: 0.030652\n","298it [00:
```
Comment (dmcguire81) [inline notebooks/run_predictions.ipynb]: Can you try to format this? I think the `json_pp` CLI tool would do it.

**Review by dmcguire81 — APPROVED** [review]
A couple of nits, but looks great, overall. Looks we're going to have pretty good results.

**Comment by dmcguire81** [comment]
> A couple of nits, but looks great, overall. Looks we're going to have pretty good results.

Feel free to merge if you do make changes and it dismisses the approval.

File: learn/training.py (hunk 2 above)
Comment (chrispebble) [reply learn/training.py]: Yes, at one point I thought this flag - which seemed to not actually be implemented correctly - might have been the reason the code would not run.  But it turned out to be unrelated.

File: learn/models.py (hunk 1 above)
Comment (chrispebble) [reply learn/models.py:8]: Ahh yes that is much more genius.

File: notebooks/dataproc_mimic_III.ipynb (hunk 3 above)
Comment (chrispebble) [reply notebooks/dataproc_mimic_III.ipynb]: Excellent point!  I was afraid to lose the output but I just saved it separately for now.

File: notebooks/run_predictions.ipynb (hunk 4 above)
Comment (chrispebble) [reply notebooks/run_predictions.ipynb]: I think this was output?  If this is still gross let me know it looked ok but I wasn't entirely sure what I was looking for.

**Pull request merged**